scripts/src/java/org/oppia/android/scripts/todo/TodoCheck.kt

```diff
@@ -0,0 +1,140 @@
+package org.oppia.android.scripts.todo
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.oppia.android.scripts.todo.data.Issue
+import java.io.File
+
+/**
+ * Script for ensuring that all TODOs present in the repository are correctly formatted and
+ * corresponds to an open issue on Github.
+ *
+ * Usage:
+ *   bazel run //scripts:todo_check -- <path_to_directory_root>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ *
+ * Example:
+ *   bazel run //scripts:todo_check -- $(pwd)
+ *
+ * NOTE TO DEVELOPERS: The script is meant only to be run during the CI, as this expects a JSON
+ *     file at the root level of the repository which contains the list of all open issues. This
+ *     file is generated during the CI. So, executing this anywhere except CI will lead to the
+ *     failure of the script.
+ */
+fun main(vararg args: String) {
+  // Path of the repo to be analyzed.
+  val repoPath = "${args[0]}/"
+
+  // Path to the JSON file containing the list of open issues.
+  val pathToOpenIssuesJsonFile = "${args[0]}/open_issues.json"
+
+  // List of all the open issues on Github for this repository.
+  val openIssueList = retrieveOpenIssueList(pathToOpenIssuesJsonFile)
+
+  val correctTodoFormatRegex = "TODO[\\(]#(\\d+)[\\)]: .+".toRegex()
```

Comment:
Why use the square brackets in this expression?

Reply:
Removed, as those were redundant.